### PR TITLE
restore: check header columns

### LINKED ISSUE #370: import successfully but checksum failed with wrong CSV header
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
Genrate data with go-tpc tpcc csv files. Then import data with tidb-lightning delimeter='"'.

2. What did you expect to see?
Import failed at the very beginning.

3. What did you see instead?
Failed after a table's checksum stage. When importing a large amount of data, this could be really frustrating for users. They'll wait for a long time to meet the failure

4. Versions of the cluster

    - TiDB-Lightning version (run `tidb-lightning -V`):

        ```
        $ ./bin/tidb-lightning -V
        Release Version: v4.0.3-9-gf18863a-dev
        Git Commit Hash: f18863a6efe8ae16c5e84b4d11094c7e485ceb38
        Git Branch: master
        UTC Build Time: 2020-08-13 06:25:22
        Go Version: go version go1.14.7 linux/amd64
        ```
        
    - TiKV-Importer version (run `tikv-importer -V`)

        ```
        (paste TiKV-Importer version here)
        ```
        
    - TiKV version (run `tikv-server -V`):

        ```
        (paste TiKV version here)
        ```

    - TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        (paste TiDB cluster version here)
        ```

    - Other interesting information (system version, hardware config, etc):

        >
        >

5. Operation logs
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs
   
6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `inventory.ini` if deployed by Ansible

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for TiDB-Lightning if possible

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Check csv header column when init chunk permutations. If there are unknown columns, return an error

close #370 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Side effects

Related changes

### CODE REVIEW

File: lightning/restore/restore_test.go [hunk 1]
```diff
@@ -344,6 +350,13 @@ func (s *tableRestoreSuite) TestPopulateChunks(c *C) {
 	err := s.tr.populateChunks(rc, cp)
 	c.Assert(err, IsNil)
 
+	for id, engine := range cp.Engines {
+		fmt.Printf("id: %d, engine: %v\n", id, *engine)
+		for _, c := range engine.Chunks {
+			fmt.Printf("  key: %v, chunk: %v\n", c.Key, c.Chunk)
+		}
+	}
+
```
Comment (kennytm) [inline lightning/restore/restore_test.go:359]: ?

File: lightning/restore/restore.go [hunk 2]
```diff
@@ -1400,19 +1404,41 @@ func (t *TableRestore) initializeColumns(columns []string, ccp *ChunkCheckpoint)
...
+
+	// check if there are some unknown columns
+	var unknownCols []string
+	for _, c := range columns {
+		if _, ok := tableColumnMap[c]; !ok && c != model.ExtraHandleName.L {
+			unknownCols = append(unknownCols, c)
+		}
+	}
+	if len(unknownCols) > 0 {
+		return colPerm, errors.Errorf("unknown columns in csv header %s", unknownCols)
```
Comment (kennytm) [inline lightning/restore/restore.go:1439]:
```suggestion
		return colPerm, errors.Errorf("unknown columns in header: %s", unknownCols)
```

this function is not CSV-specific

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by kennytm — COMMENTED** [review]
lgtm

**Comment by glorv** [comment]
the test failure is caused by https://github.com/tikv/tikv/issues/8481 this issue

**Review by 3pointer — COMMENTED** [review]
LGTM

**Review by kennytm — APPROVED** [review]

**Pull request merged**